src/MOI_wrapper.jl

```diff
@@ -940,25 +1169,84 @@ function MOI.jacobian_structure(model::Optimizer)
     return J
 end
 
+function _eval_constraint_jacobian(
+    J::AbstractVector,
+    x::AbstractVector,
+    offset::Int,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for i in 1:s.input_dimension
+        s.x[i] = x[f.variables[i].value]
+    end
+    s.jacobian(view(J, offset:length(J)), s.x)
+    for i in 1:s.output_dimension
+        J[offset+length(s.jacobian_structure)+i-1] = -1.0
+    end
+    return length(s.jacobian_structure) + s.output_dimension
+end
+
 function MOI.eval_constraint_jacobian(model::Optimizer, values, x)
     offset = MOI.eval_constraint_jacobian(model.qp_data, values, x)
+    for (f, s) in model.vector_nonlinear_oracle_constraints
+        offset += _eval_constraint_jacobian(values, x, offset, f, s)
+    end
     nlp_values = view(values, offset:length(values))
     MOI.eval_constraint_jacobian(model.nlp_data.evaluator, nlp_values, x)
     return
 end
 
 ### Eval_H_CB
 
+function _hessian_lagrangian_structure(
+    H::AbstractVector,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for (i, j) in s.hessian_lagrangian_structure
+        push!(H, (f.variables[i].value, f.variables[j].value))
+    end
+    return
+end
+
 function MOI.hessian_lagrangian_structure(model::Optimizer)
     H = MOI.hessian_lagrangian_structure(model.qp_data)
+    for (f, s) in model.vector_nonlinear_oracle_constraints
+        _hessian_lagrangian_structure(H, f, s)
+    end
     append!(H, MOI.hessian_lagrangian_structure(model.nlp_data.evaluator))
     return H
 end
 
+function _eval_hessian_lagrangian(
+    H::AbstractVector,
+    H_offset::Int,
+    x::AbstractVector,
+    μ::AbstractVector,
+    μ_offset::Int,
+    f::MOI.VectorOfVariables,
+    s::VectorNonlinearOracle,
+)
+    for i in 1:s.input_dimension
+        s.x[i] = x[f.variables[i].value]
+    end
+    H_view = view(H, H_offset:length(H))
+    μ_view = view(μ, μ_offset:length(μ))
```

Comment:
I think we need to restrict $\mu$ to only contain coordinates of the VectorNonlinearOracle's outputs:
```suggestion
    H_view = view(H, H_offset:(H_offset + length(s.hessian_lagrangian_structure) - 1))
    μ_view = view(μ, μ_offset:(μ_offset + s.output_dimension - 1))
```

Reply:
I guess you could handle this in the oracle's callbacks, since you know the input and output dimension there as well.

Reply:
Regardless, with this fixed, this works for my transient-AC networks.